### ISSUE TITLE
Config plugin bug with array of string

### ISSUE THREAD

**Issue by alexandreh2ag** [traefik/traefik#9638]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

Ref: https://github.com/alexandreh2ag/traefik-ipfilter-basicauth/issues/3

I use my own plugin https://github.com/alexandreh2ag/traefik-ipfilter-basicauth/releases/tag/v1.0.3 with config like : 

```go
type BasicAuth struct {
	Users        []string `json:"users,omitempty"`
	UsersFile    string   `json:"usersFile,omitempty"`
	Realm        string   `json:"realm,omitempty"`
	RemoveHeader bool     `json:"removeHeader,omitempty"`
	HeaderField  string   `json:"headerField,omitempty"`
}

type IPWhiteList struct {
	SourceRange []string `json:"sourceRange,omitempty"`
}

// Config the plugin configuration.
type Config struct {
	BasicAuth   BasicAuth   `json:"basicAuth,omitempty"`
	IPWhiteList IPWhiteList `json:"ipWhiteList,omitempty"`
}
```

```yml
http:
  middlewares:
    auth:
      plugin:
        auth:
          basicAuth:
            realm: "Test"
            useAuthCustomHeader: true
            headerField: "X-Test-Authorization"
            users: ["test:test"]
          ipWhiteList:
            sourceRange:
              - "192.168.1.1/27"
              - "127.0.0.1"
```
I except a value of struct Config in `func New (ctx context.Context, next http.Handler, config *Config, name string) (http.Handler, error)`: 

```yml
Config:
    IPWhiteList:
        SourceRange: []string{"192.168.1.1/27", 127.0.0.1"}
```

### What did you see instead?

I got a value of struct Config in `func New (ctx context.Context, next http.Handler, config *Config, name string) (http.Handler, error)`: 

```yml
Config:
    IPWhiteList:
        SourceRange: []string{"║24║192.168.1.1/27║127.0.0.1"}
```

I think this issue was introduce with #8885.

### What version of Traefik are you using?

This issue occurred since v2.8.2 and my plugin work with v2.8.1 and lower.

### What is your environment & configuration?

<details>
<summary>static config</summary>

```yml
accessLog: {}
api:
  dashboard: true
  insecure: true
entryPoints:
  traefik:
    address: 172.18.0.1:8080
  web:
    address: 172.18.0.1:80
experimental:
  localPlugins:
    auth:
      moduleName: github.com/alexandreh2ag/traefik-ipfilter-basicauth
global:
  checkNewVersion: false
  sendAnonymousUsage: false
log:
  level: DEBUG

providers:
  file:
    directory: ./config
    watch: true
```

</details>

<details>
<summary>dynamic config</summary>

```yaml
http:
  middlewares:
    auth:
      plugin:
        auth:
          basicAuth:
            realm: "Test"
            useAuthCustomHeader: true
            headerField: "X-Test-Authorization"
            users: ["test:test"]
          ipWhiteList:
            sourceRange:
              - "192.168.1.1/27"
              - "127.0.0.1"

  routers:
    traefik_dashboard_http:
      entrypoints:
        - web
      middlewares:
        - auth@file
      rule: "Host(`localhost`)"
      service: api@internal
```

</details>

### If applicable, please paste the log output in DEBUG level

Log provided by my code:

```
ERRO[2023-01-03T18:35:16+01:00] cannot parse CIDR whitelist [║24║192.168.1.1/27║127.0.0.1]: parsing CIDR trusted IPs <nil>: invalid CIDR address: ║24║192.168.1.1/27║127.0.0.1  routerName=traefik_dashboard_http@file entryPointName=web

```

**Comment by ldez** [comment]
Hello,

Can you provide your Traefik version?

**Comment by alexandreh2ag** [comment]
Hello @ldez,

Yes, i use v2.9.6 but the issue started with v2.8.2.

**Comment by alexandreh2ag** [comment]
Hello,

I found that the problem occurred when []string is in a nested struct.

This is not working:

```
type IPWhiteList struct {
	SourceRange []string `json:"sourceRange,omitempty"`
}

type Config struct {
	IPWhiteList IPWhiteList `json:"ipWhiteList,omitempty"`
}
```

This is working:

```
type Config struct {
	SourceRange []string `json:"sourceRange,omitempty"`
}
```

**Comment by rtribotte** [comment]
Fixed by #9671
